src/appleseed-max-impl/appleseedmetalmtl/appleseedmetalmtl.cpp

```diff
@@ -0,0 +1,884 @@
+        m_pblock->GetValue(ParamIdBumpTexmap, t, m_bump_texmap, m_params_validity);
+        m_pblock->GetValue(ParamIdBumpAmount, t, m_bump_amount, m_params_validity);
+        m_pblock->GetValue(ParamIdBumpUpVector, t, m_bump_up_vector, m_params_validity);
+
+        NotifyDependents(FOREVER, PART_ALL, REFMSG_CHANGE);
+    }
+
+    valid &= m_params_validity;
+}
+
+void AppleseedMetalMtl::Reset()
+{
+    g_appleseed_metalmtl_classdesc.Reset(this);
+
+    m_params_validity.SetEmpty();
+}
+
+Interval AppleseedMetalMtl::Validity(TimeValue t)
+{
+    Interval valid = FOREVER;
+    Update(t, valid);
+    return valid;
+}
+
+ParamDlg* AppleseedMetalMtl::CreateParamDlg(HWND hwMtlEdit, IMtlParams* imp)
+{
+    ParamDlg* param_dialog = g_appleseed_metalmtl_classdesc.CreateParamDlgs(hwMtlEdit, imp, this);
+    DbgAssert(m_pblock != nullptr);
+    update_map_buttons(m_pblock->GetMap(ParamMapIdMetal));
+    g_block_desc.SetUserDlgProc(ParamMapIdBump, new BumpParamMapDlgProc());
+    return param_dialog;
+}
+
+IOResult AppleseedMetalMtl::Save(ISave* isave)
+{
+    bool success = true;
+
+    isave->BeginChunk(ChunkFileFormatVersion);
+    success &= write(isave, FileFormatVersion);
+    isave->EndChunk();
+
+    isave->BeginChunk(ChunkMtlBase);
+    success &= MtlBase::Save(isave) == IO_OK;
+    isave->EndChunk();
+
+    return success ? IO_OK : IO_ERROR;
+}
+
+IOResult AppleseedMetalMtl::Load(ILoad* iload)
+{
+    IOResult result = IO_OK;
+
+    while (true)
+    {
+        result = iload->OpenChunk();
+        if (result == IO_END)
+            return IO_OK;
+        if (result != IO_OK)
+            break;
+
+        switch (iload->CurChunkID())
+        {
+          case ChunkFileFormatVersion:
+            {
+                USHORT version;
+                result = read<USHORT>(iload, &version);
+            }
+            break;
+
+          case ChunkMtlBase:
+            result = MtlBase::Load(iload);
+            break;
+        }
+
+        if (result != IO_OK)
+            break;
+
+        result = iload->CloseChunk();
+        if (result != IO_OK)
+            break;
+    }
+
+    return result;
+}
+
+Color AppleseedMetalMtl::GetAmbient(int mtlNum, BOOL backFace)
+{
+    return Color(0.0f, 0.0f, 0.0f);
+}
+
+Color AppleseedMetalMtl::GetDiffuse(int mtlNum, BOOL backFace)
+{
+    return m_facing_tint_color;
+}
+
+Color AppleseedMetalMtl::GetSpecular(int mtlNum, BOOL backFace)
+{
+    return Color(0.0f, 0.0f, 0.0f);
+}
+
+float AppleseedMetalMtl::GetShininess(int mtlNum, BOOL backFace)
+{
+    return 0.0f;
+}
+
+float AppleseedMetalMtl::GetShinStr(int mtlNum, BOOL backFace)
+{
+    return 0.0f;
+}
+
+float AppleseedMetalMtl::GetXParency(int mtlNum, BOOL backFace)
+{
+    return 0.0f;
+}
+
+void AppleseedMetalMtl::SetAmbient(Color c, TimeValue t)
+{
+}
+
+void AppleseedMetalMtl::SetDiffuse(Color c, TimeValue t)
+{
+    Color nc;
+    Interval iv;
+    m_pblock->SetValue(ParamIdFacingTint, t, c);
```

Comment:
What's the logic here?

Reply:
I don't remember why it's made like that. I've made it simple and applied the same thing to other materials.